common/src/main/scala/org/neo4j/spark/util/Neo4jOptions.scala

```diff
@@ -189,7 +197,7 @@ case class Neo4jQueryMetadata(query: String, queryCount: String)
 
 case class Neo4jQueryOptions(queryType: QueryType.Value, value: String)
 
-case class Neo4jSessionOptions(database: String, accessMode: AccessMode = AccessMode.READ) {
+case class Neo4jSessionOptions(database: String, var accessMode: AccessMode = AccessMode.READ) {
```

Comment:
Needs to be mutable since we don't know if we are writing or reading, and we have to set the AccessMode after we created the options


See https://github.com/neo4j-contrib/neo4j-spark-connector/pull/283/files#diff-25ce4e1a7efdf00e8d4384ee919779954bb0582152cd17869abe91629b9e8f15R18